enterprise/com/src/main/java/org/neo4j/com/storecopy/TransactionCommittingResponseUnpacker.java

```diff
@@ -328,9 +420,99 @@ public void obligation( long txId ) throws IOException
+            //      will need to be terminated, with a hint that they can simply be restarted
+            //
+            // Safe zone is a snapshotTime frame in Masters domain. All transactions that started and finished within this
+            // snapshotTime frame are guarantied to not have read any mixed state records.
+            //
+            // LARGE ASCII EXAMPLE
+            //
+            //    x---------------------------------------------------------------------------------->| TIME
+            //    |MASTER STATE
+            //    |---------------------------------------------------------------------------------->|
+            //    |                                                          Batch to apply to slave
+            //    |                                                        |<------------------------>|
+            //    |
+            //    |                                                        A
+            //    |SLAVE STATE 1 (before applying batch)         |<---T--->|
+            //    |----------------------------------------------+-------->|
+            //    |                                                        |
+            //    |                                                        |
+            //    |                                                        |      B
+            //    |SLAVE STATE 2 (mid apply)                            |<-+-T--->|
+            //    |-----------------------------------------------------+--+----->|
+            //    |                                                        |      |
+            //    |                                                        |      |
+            //    |                                                        |      |  C
+            //    |SLAVE STATE 3 (mid apply / after apply)                 |<---T-+->|
+            //    |--------------------------------------------------------+------+->|
+            //    |                                                        |      |  |
+            //    |                                                        |      |  |
+            //    |                                                        |      |  |                D
+            //    |SLAVE STATE 4 (after apply)                             |      |  |      |<---T--->|
+            //    |--------------------------------------------------------+------+--+------+-------->|
+            //
+            // * Tx start on slave when slave is in SLAVE STATE 1
+            //      - Latest applied transaction has timestamp A and safe zone is A-T.
+            //      - Tx.startTime = A
+            //
+            // Scenario 1 - Tx finish when slave is in SLAVE STATE 2
+            //      Latest applied transaction in store has timestamp B and safe zone is B-T.
+            //      Tx did not cross the safe zone boundary as Tx.startTime = A > B-T
```

Comment:
What? There is a literal line going from `A` through this transaction. How come it didn't cross any boundary?